src/main/cpp/src/parse_uri.cu

```diff
@@ -951,36 +957,105 @@ std::unique_ptr<column> parse_uri(strings_column_view const& input,
                              std::move(null_mask));
 }
 
+std::unique_ptr<column> parse_uri_ansi(strings_column_view const& input,
+                                       URI_chunks chunk,
+                                       std::optional<strings_column_view const> query_match,
+                                       bool ansi_mode,
+                                       rmm::cuda_stream_view stream,
+                                       rmm::device_async_resource_ref mr)
+{
+  // Get the result first
+  auto result = parse_uri(input, chunk, query_match, stream, mr);
+
+  // In ANSI mode, check for completely invalid URIs
+  if (ansi_mode && input.size() > 0) {
```

Comment:
This block is too big, causing difficulty for future maintenance and code review. Just make it smaller by using the other way.

```suggestion
  if (!ansi_mode || input.size() == 0) { return result; }
```